src/plugins/dashboard/public/application/embeddable/dashboard_container.tsx

```diff
@@ -95,13 +96,29 @@ export interface DashboardContainerOptions {
 export type DashboardReactContextValue = KibanaReactContextValue<DashboardContainerOptions>;
 export type DashboardReactContext = KibanaReactContext<DashboardContainerOptions>;
 
+const defaultCapabilities = {
+  show: false,
+  createNew: false,
+  saveQuery: false,
+  createShortUrl: false,
+  hideWriteControls: true,
+  showWriteControls: false,
```

Comment:
Aren't these two redundant?

Reply:
This is funny - as I deangularized this, I noticed that `showWriteControls` and `hideWriteControls` were both used in various places so I temporarily kept them.

Now looking closer, it seems like they are both stemming from the same place `application.capabilities.showWriteControls`, though hideWriteControls is managed via the `kibana_legacy` plugin, so there is potential for it to be set elsewhere. I would love to be able to unify these - though I am unsure whether it would constitute a breaking change. 

I will ask the kibana app services team about it tomorrow, but if you know of anyone else who might be able to help please let me know!

Reply:
Update - the `showWriteControls` entry here was the redundant one. I have cleaned up all usages so the legacy 'dashboard only user' should get applied properly everywhere.